packages/workspace/src/generators/move/lib/update-eslint-config.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  joinPathFragments,
+  offsetFromRoot,
+  ProjectConfiguration,
+  Tree,
+  updateJson,
+} from '@nx/devkit';
+import { join } from 'path';
+import { NormalizedSchema } from '../schema';
+
+interface PartialEsLintrcOverride {
+  parserOptions?: {
+    project?: string[];
+  };
+}
+
+interface PartialEsLintRcJson {
+  extends: string | string[];
+  overrides?: PartialEsLintrcOverride[];
+}
+
+function offsetFilePath(
+  project: ProjectConfiguration,
+  pathToFile: string,
+  offset: string
+): string {
+  if (!pathToFile.startsWith('..')) {
+    // not a relative path
+    return pathToFile;
+  }
+  const pathFromRoot = join(project.root, pathToFile);
+  return joinPathFragments(offset, pathFromRoot);
+}
+
+/**
+ * Update the .eslintrc file of the project if it exists.
+ *
+ * @param schema The options provided to the schematic
+ */
+export function updateEslintConfig(
```

Comment:
Move this to the linter package. I think it will end up much easier to work with in there.